subsys/fb/cfb.c

```diff
@@ -117,13 +157,31 @@ int cfb_print(struct device *dev, char *str, u16_t x, u16_t y)
 		return -1;
 	}
 
-	if ((fb->screen_info & SCREEN_INFO_MONO_VTILED) && !(y % 8)) {
+	if (!(y % 8)) {
 		for (size_t i = 0; i < strlen(str); i++) {
 			if (x + fptr->width > fb->x_res) {
 				x = 0U;
 				y += fptr->height;
 			}
-			x += fb->kerning + draw_char_vtmono(fb, str[i], x, y);
+			switch (fb->pixel_format) {
+			case PIXEL_FORMAT_MONO10:
+			case PIXEL_FORMAT_MONO01:
+				x += fb->kerning +
+				draw_char_vtmono(fb, str[i], x, y);
```

Comment:
I'd be happy to see this on one line.  If you really want to obey the length limits (they're not mandatory), please indent the second line to make more clear it's a continuation, rather than a mistake separating two statements.  (I also like to put the operator at the start of the second line, but you might get pushback from others on that.)

Also below.

Reply:
FYI: If you run [uncrustify](https://docs.zephyrproject.org/latest/contribute/index.html#uncrustify) on the source file it fixes style issues in a way nobody should complain about.

Reply:
ok, let me revise

Reply:
> A couple comments below. Blockers are:
> 
> * Unnecessary `CHARACTER_FRAMEBUFFER_USE_CST_FONTS` Kconfig
> * Overcomplex expression in `draw_char_vtrgb565`
> * Indentation for split statements
> 
> I also need you to document where to get the new font, and how to regenerate the glyphs. I found the font, and used the updated script in #17541, but can't replicate the bit data, which may be because of changes in that PR or because it was generated with some other tool. IMO you don't have to use that script, but you do need to document how you generated the data: if somebody wants to use the font but needs a different size or weight, they'll want this information.
> 
> FWIW I would like to see this merged soon, as I have work in progress that will need to be rebased on (and can take advantage of) these changes.

I download the font from https://www.download-free-fonts.com/ and the license can be fond in the header file of cfb_fonts8x8.c, and I use on line converter to c array, you can find it here
https://littlevgl.com/ttf-font-to-c-array

Reply:
Please put that in a commit message that adds the font.  We can't expect people to track it down in the github review archives.

Reply:
ok